pkg/os/os.go

```diff
@@ -90,7 +91,11 @@ func (RealOS) Mount(source string, target string, fstype string, flags uintptr,
 	return unix.Mount(source, target, fstype, flags, data)
 }
 
-// Unmount will call unix.Unmount to unmount the file.
+// Unmount will call unix.Unmount to unmount the file. The function doesn't
+// return error if target is not mounted.
 func (RealOS) Unmount(target string, flags int) error {
+	if mounted, err := mount.Mounted(target); err != nil || !mounted {
```

Comment:
be careful with the check here, if shm path is under `/var/run/...` but the mount in the `/proc/self/mountinfo` is in the form of `/run/...` you'll get `mounted = false, err = nil`.
For instance, in CRI-O we store shm under `/var/run/CONTID/shm` but if you pass that to `mount.Mounted` it'll say that's not mounted because `/proc/self/mountinfo` only contains `/run/CONTID/shm` and that doesn't match.
We do resolve the symlink first.